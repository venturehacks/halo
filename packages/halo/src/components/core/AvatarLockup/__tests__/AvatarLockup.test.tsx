import { render } from 'enzyme';
import React from 'react';

import { AvatarLockup } from '../index';

const sharedProps = {
  children: <>Vishal</>,
  imageUrl: 'https://avatars.githubusercontent.com/u/56262247',
};

describe('Avatar', () => {
  test('smoke', () => {
    const component = render(<AvatarLockup {...sharedProps} />);

    expect(component.html()).toContain('<img');
  });

  describe('snapshots', () => {
    describe('size', () => {
      test(`sm`, () => {
        const component = render(<AvatarLockup {...sharedProps} size="sm" />);

        expect(component).toMatchSnapshot();
      });

      test(`md [default]`, () => {
        const component = render(<AvatarLockup {...sharedProps} />);

        expect(component).toMatchSnapshot();
      });

      test(`lg`, () => {
        const component = render(<AvatarLockup {...sharedProps} size="lg" />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('orientation', () => {
      test(`left [default]`, () => {
        const component = render(<AvatarLockup {...sharedProps} />);

        expect(component).toMatchSnapshot();
      });
      test(`right`, () => {
        const component = render(
          <AvatarLockup {...sharedProps} orientation="right" />,
        );

        expect(component).toMatchSnapshot();
      });

      test(`top`, () => {
        const component = render(
          <AvatarLockup {...sharedProps} orientation="top" />,
        );

        expect(component).toMatchSnapshot();
      });
    });
  });
});
