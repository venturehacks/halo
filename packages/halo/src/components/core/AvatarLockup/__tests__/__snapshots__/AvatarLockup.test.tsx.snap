// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar snapshots orientation left [default] 1`] = `
<div
  class="flex antialiased flex-row items-center gap-3"
>
  <div
    class="order-1 inline-flex flex-row items-center relative border border-gray-400 bg-gray-100 rounded-full h-12 w-12 order-1"
  >
    <img
      alt="Avatar"
      class="rounded-full"
      height="46"
      src="https://avatars.githubusercontent.com/u/56262247"
      width="46"
    />
  </div>
  <div
    class="order-2"
  >
    Vishal
  </div>
</div>
`;

exports[`Avatar snapshots orientation right 1`] = `
<div
  class="flex antialiased flex-row items-center gap-3"
>
  <div
    class="order-3 inline-flex flex-row items-center relative border border-gray-400 bg-gray-100 rounded-full h-12 w-12 order-3"
  >
    <img
      alt="Avatar"
      class="rounded-full"
      height="46"
      src="https://avatars.githubusercontent.com/u/56262247"
      width="46"
    />
  </div>
  <div
    class="order-2"
  >
    Vishal
  </div>
</div>
`;

exports[`Avatar snapshots orientation top 1`] = `
<div
  class="flex antialiased flex-col gap-3"
>
  <div
    class="order-1 inline-flex flex-row items-center relative border border-gray-400 bg-gray-100 rounded-full h-12 w-12 order-1"
  >
    <img
      alt="Avatar"
      class="rounded-full"
      height="46"
      src="https://avatars.githubusercontent.com/u/56262247"
      width="46"
    />
  </div>
  <div
    class="order-2"
  >
    Vishal
  </div>
</div>
`;

exports[`Avatar snapshots size lg 1`] = `
<div
  class="flex antialiased flex-row items-center gap-4"
>
  <div
    class="order-1 inline-flex flex-row items-center relative border border-gray-400 bg-gray-100 rounded-full h-18 w-18 order-1"
  >
    <img
      alt="Avatar"
      class="rounded-full"
      height="70"
      src="https://avatars.githubusercontent.com/u/56262247"
      width="70"
    />
  </div>
  <div
    class="order-2"
  >
    Vishal
  </div>
</div>
`;

exports[`Avatar snapshots size md [default] 1`] = `
<div
  class="flex antialiased flex-row items-center gap-3"
>
  <div
    class="order-1 inline-flex flex-row items-center relative border border-gray-400 bg-gray-100 rounded-full h-12 w-12 order-1"
  >
    <img
      alt="Avatar"
      class="rounded-full"
      height="46"
      src="https://avatars.githubusercontent.com/u/56262247"
      width="46"
    />
  </div>
  <div
    class="order-2"
  >
    Vishal
  </div>
</div>
`;

exports[`Avatar snapshots size sm 1`] = `
<div
  class="flex antialiased flex-row items-center gap-1"
>
  <div
    class="order-1 inline-flex flex-row items-center relative border border-gray-400 bg-gray-100 rounded-full h-8 w-8 order-1"
  >
    <img
      alt="Avatar"
      class="rounded-full"
      height="30"
      src="https://avatars.githubusercontent.com/u/56262247"
      width="30"
    />
  </div>
  <div
    class="order-2"
  >
    Vishal
  </div>
</div>
`;
